### ISSUE TITLE
Update testing styleguide to inform that todo tests should not be merged

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7819]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#7819|
|Type|enhancement|
|Created|2015-03-16 16:49:01Z|
|Branch|https://github.com/twisted/twisted/tree/test-todo-documentation-7819|

We should update this document
http://twistedmatrix.com/documents/current/core/development/policy/test-standard.html

<details><summary>Searchable metadata</summary>

```
trac-id__7819 7819
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__ 
branch__branches_test_todo_documentation_7819 branches/test-todo-documentation-7819
branch_author__adiroiban adiroiban
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1426524541670213 1426524541670213
changetime__1426978360959348 1426978360959348
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

(In [44069]) Branching to test-todo-documentation-7819.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban removed owner|
|-|-|

I have pushed an initial change.

Please review.

It explicitly ask not to commit .todo tests but create a ticket for them.

It tells you that is OK if you do TDD without .todo markers. 

BTW do you still use .todo markers as a day to day development practice?
I just write the test to see it fail and don't bother with the marker.

I have also change the hierarchy for sections so that Skipping Tests / Writing New Tests / Test Coverate are same level sections.

-------

While on this what do you think if we add xUnit Test Patterns: Refactoring Test Code by Gerard Meszaros to the list of links ? I find it a good book.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl set owner to @adiroiban|
|-|-|

Thanks, Adi.

- docs/core/development/policy/test-standard.rst:440,441,443 -- "test" here should be "tests"
- docs/core/development/policy/test-standard.rst:441 -- "and add the test to ticket's description" -> "and add the test to the ticket's description."

Please fix up these two issues and merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set status to `closed`|
|-|-|

(In [44177]) Merging test-todo-documentation-7819: Update testing styleguide to inform that todo tests should not be merged.

Author: adiroiban
Reviewers: hawkowl
Fixes: [#7819](https://github.com/twisted/twisted/issues/7819)
